Framework/API/src/ISpectrum.cpp

```diff
@@ -135,6 +135,22 @@ bool ISpectrum::hasDx() const { return bool(histogramRef().sharedDx()); }
  */
 void ISpectrum::resetHasDx() { mutableHistogramRef().setSharedDx(nullptr); }
 
+/**
+ * Makes sure the histogram has valid Y data.
+ * @param histogram A histogram to process.
+ * @throw std::invalid_argument if Y data is NULL.
+ */
+void ISpectrum::checkAndSanitizeHistogram(HistogramData::Histogram &histogram) {
```

Comment:
Originally I suggested it because the interface in the base class `ISpectrum` directly constructs the Histogram [here](https://github.com/mantidproject/mantid/blob/master/Framework/API/inc/MantidAPI/ISpectrum.h#L104) but relies on the derived class to provide the override for any checks to happen. 
Imagining we had future workspace types which inherited `ISpectrum` would not having Y and/or E data be an exception to the rule or something that is worth checking on a case by case basis?